src/rdf4cpp/rdf/Node.cpp

```diff
@@ -154,6 +174,9 @@ std::weak_ordering Node::operator<=>(const Node &other) const noexcept {
 }
 
 bool Node::operator==(const Node &other) const noexcept {
+    if (!this->is_literal() && !other.is_literal() && this->backend_handle().node_storage_id() == other.backend_handle().node_storage_id()) {
```

Comment:
Does this not work for literals? Could you explain why, I think I am missing somthing

Reply:
Because Literals of different datatypes can be equals, which gives us different backend handles even if they are in the same node storage.

Reply:
Right, forgot that. Could you add a comment explaining that?